crates/primitives/src/withdrawal.rs

```diff
@@ -33,6 +37,73 @@ impl Withdrawal {
     }
 }
 
+/// Represents a collection of Withdrawals.
+#[main_codec]
+#[derive(Debug, Clone, PartialEq, Eq, Default, Hash)]
+pub struct Withdrawals(Vec<Withdrawal>);
+
+impl Withdrawals {
+    /// Create a new Withdrawals instance.
+    pub fn new(withdrawals: Vec<Withdrawal>) -> Self {
+        Withdrawals(withdrawals)
+    }
+
+    /// Calculate a heuristic for the in-memory size of the [Withdrawals].
+    #[inline]
+    pub fn size(&self) -> usize {
+        self.iter().map(Withdrawal::size).sum()
+    }
+
+    /// Get an iterator over the Withdrawals.
+    pub fn iter(&self) -> std::slice::Iter<'_, Withdrawal> {
+        self.0.iter()
+    }
+
+    /// Get a mutable iterator over the Withdrawals.
+    pub fn iter_mut(&mut self) -> std::slice::IterMut<'_, Withdrawal> {
+        self.0.iter_mut()
+    }
+}
+
+impl IntoIterator for Withdrawals {
+    type Item = Withdrawal;
+    type IntoIter = std::vec::IntoIter<Withdrawal>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+impl Deref for Withdrawals {
+    type Target = Vec<Withdrawal>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for Withdrawals {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl Encodable for Withdrawals {
+    fn encode(&self, out: &mut dyn BufMut) {
+        self.0.encode(out);
+    }
+
+    fn length(&self) -> usize {
+        self.0.length()
+    }
+}
+
+impl Decodable for Withdrawals {
+    fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
+        Ok(Self(Decodable::decode(buf)?))
+    }
+}
```

Comment:
This can be done with `RlpEncodableWrapper` and `RlpDecodableWrapper`

Reply:
Nice thanks, just fixed, didn't know that